content/buildah-podman-skopeo.howto.md

```diff
@@ -0,0 +1,125 @@
+---
+title: "Building Containers safely without Docker (Look Ma! No Docker!)"
+date: 2020-09-29T13:42:36-07:00 
+draft: true
+---
+
+If you want to create a container, the traditional way to go about it is to use just `docker build .` and let docker take care of it for you.
+
+One problem with this is that it requires access to the Docker daemon which then runs as root. It also tends to lead to more bloated images as the typical Dockerfile starts with pulling in someone elses' base container (not always bad, just be aware of where your base containers come from).
+
+These problems become more important when you start automating container builds as someone could use a CI/CD pipeline to execute their own code which would get run with elevated privileges! Removing the dependency on the Daemon removes this path of exploits.
+
+
+My preferred way to build my container images is with [Buildah](https://buildah.io)!
+
+Note: I'll probably be using "container" and "container image" interchangeably. It should be obvious from context what I mean.
+
+Bulidah uses the standard CRI set of tooling and configuration which means that it runs very nicely along side CRI-O, Podman, and Skopeo making for a full (mostly drop in) replacement for Docker.
```

Comment:
typo: Bulidah -> Buildah

Reply:
Sorry, tagged before the push was done apparently